python/app/model_entity_listing.py

```diff
@@ -190,3 +196,50 @@ def _populate_thumbnail_image(self, item, field, image, path):
         sg_data = item.get_sg_data()
         icon = self._sg_formatter.create_thumbnail(image, sg_data)
         item.setIcon(QtGui.QIcon(icon))
+
+    def _before_data_processing(self, data):
+        """
+        Called just after data has been retrieved from Shotgun but before any
+        processing takes place.
+        """
+
+        self.content_is_partial = len(data) >= self.my_record_limit
+        if self.content_is_partial:
+            data = data[:-1]
+
+        return data
+
+    def _format_partial_tooltip(self, text):
+        """
+        Used by children classes to alter the text.
+        """
+        return text
+
+    ############################################################################################
+    # private methods
+
+    def _on_data_updated(self):
+        if not self.label_partial:
+            return
+
+        self.label_partial.setVisible(self.rowCount() > 0)
+
+        text = "{num} items loaded"
+        tooltip = "This is the number of items loaded from ShotGrid."
+        if self.content_is_partial:
+            text += " (partial result)"
+            tooltip = self._format_partial_tooltip(
+                "This panel only loads a maximum of {num} items from ShotGrid."
+            )
```

Comment:
My friendly suggestion is to declare a constant var for the messages strings at the beginning of the class